### DISCUSSION TITLE
I like it

### DISCUSSION THREAD

**Discussion by mikkelking** [hongaar/bandersnatch#318] [Q&A]
I like the look of this, and am about to start building a REPL for my app with it.

I see bandersnatch was inspired by vorpal, which now seems to be abandoned. You appear to have added features, like options. Any more background info available on your motivation and a possible roadmap?

I think I will use it in an interactive way, so for example I will provide commands to 

- select an object (and display info about the selected object)
- do an operation on that object (and display a summary of the operation)
- confirm the operation with a "go" command

This appears to be quite feasible to do. I may have some questions or feature requests along the way.

**Comment by hongaar** [comment]
Hi @mikkelking, appreciate the feedback! I started writing this library after porting vorpal to TypeScript and realising the codebase was a mess. Some of that work and discussions can be found here: https://github.com/vorpaljs-reforged/vorpal/.

What you're describing should be possible with bandersnatch. If there's anything I can help you with or clarify about this library, let me know here! And would be interested to hear about any features you find missing from this package.

**Comment by kevinkhill** [comment]
Hiya, I just found your library too and I also like it 😃 I just started, but I think it will be a good fit.

If you want to start a "used in" listing, then [its in here](https://github.com/cnc4me/cnc4me/blob/main/apps/macro-cli/src/cli.ts)

**Comment by jq170727** [comment]
Just adding a note to say I've started using this for interactive testing.   I also tried vorpal for a bit but really wanted something simpler and more Typescript focused so I kept searching and eventually came across this module.  There are a few rough spots but so far it's working very well for me.  Thank you for creating this!

It would be nice if [.save and .load behave as in the Node.js repl](https://github.com/hongaar/bandersnatch/discussions/443).  I haven't dug into it too deep yet but from what I can discern bandersnatch is mediating between yargs, enquirer and part of the Node.js repl so I suspect getting .save/load working correctly could be easy or hard depending on what those packages apis let you do.  I might try and fix this on my own if I can figure out how everything works.
